### ISSUE TITLE
Can't Integrate sqrt(1 - cos(x)^2)

### ISSUE THREAD

**Issue by weakit** [sympy/sympy#19738]
```python
integrate(sqrt(1 - cos(x)**2), x)
```
never returns.
You get the same result with sqrt(1 ± sin(x)^2).

This however works:
```python
>>> trigsimp(sqrt(1 - sin(x)**2))
sqrt(cos(x)**2)
>>> integrate(_, x)
sqrt(cos(x)**2)*sin(x)/cos(x)
```

I'm not sure but maybe this is related to #15730

**Comment by oscargus** [comment]
Maybe a solution here would be to check if an unevaluated integral has trigonometric functions in it and if so rerun the intergral after simplification (assuming that the simplified expression is not identical to the previous).

**Comment by weakit** [comment]
> rerun the intergral after simplification

@oscargus wouldn't most integrals finish faster and produce better results if the input is cleaned up and simplified first?

**Comment by oscargus** [comment]
@weakit it may very well be so. However, in general simplify can be quite time consuming and there are probably cases where one can "trick" the integration to work out by writing it in a particular form, so I am not sure it is a good idea to just always perform it. I've found at least three integrals (two from the issues and one from a standard test suite that we run) that will be solved by executing `trigsimp`, although, as you say, there may be other cases one can identify as well.

**Comment by weakit** [comment]
Then I would assume your original idea of simplifying trigonometric functions would be better. Even though it might take some time, it would certainly be better than not returning.

**Comment by oscargus** [comment]
Indeed! However, it would not make anyone happy if it first takes a long time not to evaluate, then long time to simplify just to take a long time not to evaluate the simplified expression... I'm collecting cases in #22081 so that one can see how common this is (I cannot seem to find the second issue for trigsimp although I am quite sure there is one...).

I've browsed the code, but the way it is written, it is not obvious (to me) where to add things. Otherwise it would be a simple as:
```
if integral.has(Integral):   # Probably one should identify the actual remaining integral in case parts have been solved
    if integral.has(TrigonometricFunctions):
        integrate(trigsimp(integral))
```
and so on.

**Comment by weakit** [comment]
I don't think you should be simplifying any expression that has a trig function, just the ones that match this pattern.

I think it would be better to add a case for these trig functions and have only the required parts re-written and simplified. I'm not very familiar with how sympy manages to evaluate integrals however, so good luck.
